### PR TITLE
Fix smart coupons on redirect flow & display available coupons on embedded checkout page

### PR DESCRIPTION
**Pull request by mntzrr**
https://app.clickup.com/t/86934ur9d

### CODE REVIEW

File: templates/avarda-checkout.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 
 do_action( 'aco_wc_before_checkout_form' );
+do_action( 'woocommerce_before_checkout_form', WC()->checkout() );
```
Comment (MichaelBengtsson) [inline templates/avarda-checkout.php:12]: Did we discuss this internally to say this is a OK solution? Wont this possibly make the giftcards show twice for people that have manually added this action themselves?

File: composer.json [hunk 2]
```diff
@@ -5,17 +5,13 @@
     "require-dev": {
         "php-stubs/woocommerce-stubs": "^8.5",
         "wp-coding-standards/wpcs": "^2.3",
-        "wpify/scoper": "3.1.3"
+        "wpify/scoper": "3.1.3",
+        "squizlabs/php_codesniffer": "*"
```
Comment (MichaelBengtsson) [inline composer.json:9]: Why is squizlabs/php_codesniffer added as a separate dependency? Its included as a dependency from [wp-coding-standards/wpcs](https://github.com/WordPress/WordPress-Coding-Standards/blob/develop/composer.json#L24) so should not be required to be added directly right?

File: classes/requests/helpers/class-aco-helper-cart.php [hunk 3]
```diff
@@ -57,32 +57,21 @@ public function get_cart_items( $cart = null ) {
...
-					$label        = apply_filters( 'aco_smart_coupon_gift_card_label', esc_html( sprintf( __( 'Gift card: %s', 'avarda-checkout-for-woocommerce' ), $coupon->get_code() ) ), $coupon );
-					$giftcard_sku = apply_filters( 'aco_smart_coupon_gift_card_sku', esc_html( $coupon->get_id() ), $coupon );
-					$gift_card    = array(
-						'notes'       => $giftcard_sku,
-						'description' => substr( $label, 0, 34 ),
-						'quantity'    => 1,
-						'amount'      => $coupon_amount,
-						'taxCode'     => 0,
-						'taxAmount'   => 0,
-					);
```
Comment (MichaelBengtsson) [inline classes/requests/helpers/class-aco-helper-cart.php:82]: Are we removing support for Smart Coupons? I only see the WCGiftCards plugins compatibility class being used.

File: classes/requests/helpers/class-aco-helper-order.php [hunk 4]
```diff
@@ -324,31 +324,21 @@ public static function get_reference( $order_item ) {
...
-						'notes'       => $giftcard_sku,
-						'description' => substr( $label, 0, 34 ),
-						'quantity'    => 1,
-						'amount'      => $coupon_amount,
-						'taxCode'     => 0,
-						'taxAmount'   => 0,
-					);
-
-					$items[] = $gift_card;
-				}
```
Comment (MichaelBengtsson) [inline classes/requests/helpers/class-aco-helper-order.php:351]: Are we removing support for Smart Coupons? I only see the WCGiftCards plugins compatibility code being used.

**Review by MichaelBengtsson — CHANGES_REQUESTED** [review]

File: classes/requests/helpers/class-aco-helper-cart.php (hunk 3 above)
Comment (MichaelBengtsson) [reply classes/requests/helpers/class-aco-helper-cart.php:82]: Ignore this, i read the code wrong

File: classes/requests/helpers/class-aco-helper-order.php (hunk 4 above)
Comment (MichaelBengtsson) [reply classes/requests/helpers/class-aco-helper-order.php:351]: Ignore this, read the code wrong

File: templates/avarda-checkout.php (hunk 1 above)
Comment (mntzrr) [reply templates/avarda-checkout.php:12]: This was not intentional. I had it to display the coupon on the checkout page. I'll remove it.

**Review by MichaelBengtsson — APPROVED** [review]

**Pull request merged**